### PR TITLE
disco_l475_iot: Enable peripheral sample 

### LINKED ISSUE #15714: samples/bluetooth/peripheral: could not connect with disco_l475_iot1 board
**Describe the bug**
When I tried to run `samples/bluetooth/peripheral/` in `disco_l475_iot1` board and unable to connect with Bluez. The sample application crashes with exception.

**To Reproduce**
1. cd samples/bluetooth/peripheral; mkdir build; cd build
2. cmake -GNinja -DBOARD=disco_l475_iot1 ..
3. ninja; ninja flash
4. Run bluetoothctl: trust the zephyr device and pair it.

**Expected behavior**
Zephyr and Bluetoothctl should ideally pair and connect.

**Impact**
This blocks the basic bluetooth functional testing and smp_svr use case on top of mcuboot.

**Screenshots or console output**
```
***** Booting Zephyr OS v1.14.0-rc1-2148-g3ce677092d36 *****
[00:00:00.005,000] <inf> mcuboot: Starting bootloader
[00:00:00.012,000] <inf> mcuboot: Primary image: magic=unset, copy_done=0x3, image_ok=0x3
[00:00:00.020,000] <inf> mcuboot: Scratch: magic=unset, copy_done=0x0, image_ok=0x3
[00:00:00.028,000] <inf> mcuboot: Boot source: primary slot
[00:00:00.049,000] <inf> mcuboot: Swap type: none
[00:00:00.251,000] <inf> mcuboot: Bootloader chainload address offset: 0x20000
[00:00:00.258,000] <inf> mcuboot: Jumping to the first image slot
***** Booting Zephyr OS v1.14.0-rc1-2150-g0dd3b421c4c4 *****
[00:00:00.024,000] <dbg> bt_gatt.sc_process: sBluetooth initialized
tart 0Advertising successfully started
x0010 end 0x0016
--- 528 messages dropped ---
[00:00:00.259,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0018 offset 0 length 5
[00:00:00.259,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x001b offset 0 length 5
[00:00:00.260,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x001d offset 0 length 5
[00:00:00.261,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x001f offset 0 length 2
[00:00:00.261,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0020 offset 0 length 5
[00:00:00.262,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0023 offset 0 length 2
[00:00:00.262,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0024 offset 0 length 5
[00:00:00.263,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0027 offset 0 length 16
[00:00:00.264,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0028 offset 0 length 16
[00:00:00.265,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x002b offset 0 length 16
[00:00:00.266,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x002d offset 0 length 16
[00:00:00.267,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x002f offset 0 length 2
[00:00:00.267,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0030 offset 0 length 16
[00:00:00.268,000] <dbg> bt_gatt.bt_gatt_attr_read: handle 0x0032 offset 0 length 16
[00:00:00.270,000] <dbg> bt_gatt: Hash: 
9d b0 23 d1 d5 24 27 e2 |..#..$'.
a9 36 0b 9b 5c 3b f5 78 |.6..\;.x
[00:00:00.321,000] <dbg> bt_gatt.db_hash_store: Database Hash stored
Connected
[00:00:20.661,000] <dbg> bt_hci_core.bt_recv: buf 0x200030c0 len 21
[00:00:20.661,000] <dbg> bt_hci_core.hci_event: event 0x3e
[00:00:20.661,000] <dbg> bt_hci_core.hci_le_meta_event: subevent 0x01
[00:00:20.661,000] <dbg> bt_hci_core.le_legacy_conn_complete: status 0 role 1 b8:27:eb:93:88:8c (public)
[00:00:20.662,000] <dbg> bt_hci_core.enh_conn_complete: status 0 handle 2049 role 1 b8:27:eb:93:88:8c (public)
[00:00:20.662,000] <dbg> bt_conn.bt_conn_set_state: disconnected -> connected
[00:00:20.662,000] <dbg> bt_conn.bt_conn_ref: handle 2049 ref 2
[00:00:20.662,000] <dbg> bt_att.bt_att_accept: conn 0x200005ec handle 2049
[00:00:20.662,000] <dbg> bt_l2cap.bt_l2cap_chan_add: conn 0x200005ec chan 0x20000720
[00:00:20.662,000] <dbg> bt_att.bt_att_connected: chan 0x20000720 cid 0x0004
[00:00:20.662,000] <dbg> bt_l2cap.bt_l2cap_chan_add: conn 0x200005ec chan 0x200008c0
[00:00:20.662,000] <dbg> bt_l2cap.l2cap_accept: conn 0x200005ec handle 2049
[00:00:20.662,000] <dbg> bt_l2cap.bt_l2cap_chan_add: conn 0x200005ec chan 0x2000068c
[00:00:20.662,000] <dbg> bt_l2cap.l2cap_connected: ch 0x2000068c cid 0x0005
[00:00:20.663,000] <dbg> bt_gatt.bt_gatt_connected: conn 0x200005ec
[00:00:20.663,000] <dbg> bt_conn.bt_conn_unref: handle 2049 ref 1
[00:00:20.664,000] <dbg> bt_hci_core.process_events: count 2
[00:00:20.664,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:20.664,0Called legacy_request_tk
Passkey for b8:27:eb:93:88:8c (public): 207703
0[00:00:20.819,000] <dbg> bt_conn.bt_conn_send_cb: conn handle 2049 buf len 7 cb 0x080292b5
--- 30 messages dropped ---
[00:00:20.819,000] <dbg> bt_att.att_mtu_req: Negotiated MTU 65
[00:00:20.819,000] <dbg> bt_conn.bt_conn_unref: handle 2049 ref 1
[00:00:20.819,000] <dbg> bt_hci_core.process_events: count 4
[00:00:20.819,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:20.819,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:20.819,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:20.819,000] <dbg> bt_hci_core.process_events: ev->state 4
[00:00:20.819,000] <dbg> bt_conn.bt_conn_process_tx: conn 0x200005ec
[00:00:20.819,000] <dbg> bt_conn.send_buf: conn 0x200005ec buf 0x20003108 len 11
[00:00:20.819,000] <dbg> bt_conn.send_frag: conn 0x200005ec buf 0x20003108 len 11 flags 0x00
[00:00:20.819,000] <dbg> bt_conn.bt_conn_notify_tx: conn 0x200005ec
[00:00:20.819,000] <dbg> bt_conn.add_pending_tx: conn 0x200005ec cb 0x00000000
[00:00:20.820,000] <dbg> bt_hci_core.bt_send: buf 0x20003108 len 15 type 2
[00:00:20.820,000] <dbg> bt_conn.bt_conn_prepare_events: 
[00:00:20.820,000] <dbg> bt_conn.bt_conn_prepare_events: Adding conn 0x200005ec to poll list
[00:00:20.820,000] <dbg> bt_hci_core.hci_tx_thread: Calling k_poll with 4 events
[00:00:20.820,000] <dbg> bt_hci_core.process_events: count 4
[00:00:20.820,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:20.820,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:20.820,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:20.821,000] <dbg> bt_hci_core.process_events: ev->state 4
[00:00:20.821,000] <dbg> bt_conn.bt_conn_process_tx: conn 0x200005ec
[00:00:20.821,000] <dbg> bt_conn.send_buf: conn 0x200005ec buf 0x20003120 len 7
[00:00:20.821,000] <dbg> bt_conn.send_frag: conn 0x200005ec buf 0x20003120 len 7 flags 0x00
***** USAGE FAULT *****
  Illegal load of EXC_RETURN into PC
***** Hardware exception *****
Current thread ID = 0x2000021c
Faulting instruction address = 0x3c6ef372
Fatal fault in thread 0x2000021c! Aborting.
[00:00:21.020,000] <dbg> bt_hci_core.hci_num_completed_packets: num_handles 1
[00:00:21.027,000] <dbg> bt_hci_core.hci_num_completed_packets: handle 2049 count 1
[00:00:21.035,000] <dbg> bt_conn.bt_conn_ref: handle 2049 ref 2
[00:00:21.042,000] <dbg> bt_conn.bt_conn_unref: handle 2049 ref 1
[00:00:21.048,000] <dbg> bt_conn.bt_conn_notify_tx: conn 0x200005ec
[00:00:21.055,000] <dbg> bt_conn.add_pending_tx: conn 0x200005ec cb 0x080292b5
[00:00:21.062,000] <dbg> bt_hci_core.bt_send: buf 0x20003120 len 11 type 2
[00:00:21.070,000] <dbg> bt_conn.bt_conn_prepare_events: 
[00:00:21.076,000] <dbg> bt_conn.bt_conn_prepare_events: Adding conn 0x200005ec to poll list
[00:00:21.085,000] <dbg> bt_hci_core.hci_tx_thread: Calling k_poll with 4 events
***** USAGE FAULT *****
  Illegal load of EXC_RETURN into PC
***** Hardware exception *****
Current thread ID = 0x20000494
Faulting instruction address = 0x20004080
Fatal fault in thread 0x20000494! Aborting.
[00:00:25.664,000] <dbg> bt_conn.conn_le_update_timeout: conn 0x200005ec
[00:00:25.671,000] <dbg> bt_conn.send_conn_le_param_update: conn 0x200005ec features 0x00 params (24-40 0 42)
[00:00:25.681,000] <dbg> bt_l2cap.bt_l2cap_send_cb: conn 0x200005ec cid 5 len 12
[00:00:25.689,000] <dbg> bt_conn.bt_conn_send_cb: conn handle 2049 buf len 16 cb 0x00000000
[00:00:25.698,000] <dbg> bt_hci_core.process_events: count 4
[00:00:25.704,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:25.710,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:25.717,000] <dbg> bt_hci_core.process_events: ev->state 0
[00:00:25.723,000] <dbg> bt_hci_core.process_events: ev->state 4
[00:00:25.729,000] <dbg> bt_conn.bt_conn_process_tx: conn 0x200005ec
[00:00:25.736,000] <dbg> bt_conn.send_buf: conn 0x200005ec buf 0x20003120 len 16
[00:00:25.744,000] <dbg> bt_conn.send_frag: conn 0x200005ec buf 0x20003120 len 16 flags 0x00
[00:00:50.806,000] <err> bt_smp: SMP Timeout
```

**Environment (please complete the following information):**
 - OS: Raspverry PI 3 B+ for bluetoothctl
 - SDK Version: 0.10.0
 - Commit SHA: 0dd3b421c4c4338fb013416e591b97c031c4bcf3

**Additional context**
Raspberrpi bluetoothctl pairing info:
```
[bluetooth]# pair 4C:5E:AA:B4:67:EC 
Attempting to pair with 4C:5E:AA:B4:67:EC
[CHG] Device 4C:5E:AA:B4:67:EC Connected: yes
Request passkey
[agent] Enter passkey (number in 0-999999): 207703
[Zephyr Peripheral Sample Long]# info 
Device 4C:5E:AA:B4:67:EC (random)
	Name: Zephyr Peripheral Sample Long
	Alias: Zephyr Peripheral Sample Long
	Paired: no
	Trusted: yes
	Blocked: no
	Connected: yes
	LegacyPairing: no
	UUID: Heart Rate                (0000180d-0000-1000-8000-00805f9b34fb)
	UUID: Battery Service           (0000180f-0000-1000-8000-00805f9b34fb)
	UUID: Current Time Service      (00001805-0000-1000-8000-00805f9b34fb)
	UUID: Vendor specific           (12345678-1234-5678-1234-56789abcdef0)
	RSSI: -51
	AdvertisingFlags:
  06                                               .               
[CHG] Device 4C:5E:AA:B4:67:EC Connected: no
Failed to pair: org.bluez.Error.ConnectionAttemptFailed
[DEL] Device 4C:5E:AA:B4:67:EC Zephyr Peripheral Sample Long
```

### PR DESCRIPTION
**Pull request by erwango**
This PR enables peripheral sample on disco_l475_iot1.
Also for convenience, it proposes a default configuration when using btmon

Fixes #15714

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
@erwango I already approved, but it'd be good if you could also add the answer to the question "why?" to the ATT flow enforcement patch commit message. Thanks.

File: boards/arm/disco_l475_iot1/Kconfig.defconfig [hunk 1]
```diff
@@ -64,6 +64,13 @@ config UART_1
 
 endif # UART_CONSOLE
 
+if BT_DEBUG_MONITOR
+
+config UART_4
+	default y
+
+endif # UART_CONSOLE
```
Comment (frantony) [inline boards/arm/disco_l475_iot1/Kconfig.defconfig]: UART_CONSOLE -> BT_DEBUG_MONITOR?

**Comment by jhedberg** [comment]
@erwango seems like this is missing a Fixes #15714 in the description?

**Review by jhedberg — CHANGES_REQUESTED** [review]
Marking as "Request changes" until we get some better reasoning on the ATT flow control disabling (more details in my comment in #15714).

**Comment by erwango** [comment]
Closing as issue has been properly fixed by #16748.
Will push a new PR for the monitor thing.

**Pull request closed without merging**